Modules/database_manager.py

```diff
@@ -0,0 +1,159 @@
+        return cls._instances[cls]
+
+
+# noinspection SqlNoDataSourceInspection
+class DatabaseManager(metaclass=Singleton):
+
+    def __init__(self):
+        """
+        Connects to the DB, sets up the connection, retrieves the cursor.
+        Creates the default tables should they not exist.
+        """
+        # connect to PostgreSQL (PSQL) database
+        __config: dict = config.config.get("database")
+        connect_str = ("Driver={PostgreSQL UNICODE};"
+                       f"Server={__config.get('server')};"
+                       f"Port={__config.get('port')};"
+                       f"Database={__config.get('database')};"
+                       f"Uid={__config.get('username')};"
+                       f"Pwd={__config.get('password')};"
+                       "MaxVarcharSize=1024 * 1024 * 1024"
+                       # Allow bigger VarcharSize to allow faster interaction
+                       )
+        self.connection: pyodbc.Connection = pyodbc.connect(connect_str, autocommit=True)
+        self.connection.setdecoding(pyodbc.SQL_WCHAR, encoding='utf-8')
+        self.connection.setencoding(encoding='utf-8')
+        self.connection.maxwrite = 1024 * 1024 * 1024  # Again, increase this for faster access
+        self.cursor = self.connection.cursor()
+        # Ensure all the default tables are defined
+        self.cursor.execute("CREATE TABLE IF NOT EXISTS"
+                            " mecha3_facts (name VARCHAR, lang VARCHAR, response VARCHAR);")
+
+    async def _select_rows(self, table_name: str, connector: str, condition: dict = None,
+                           skipdouble_dash_test = False) -> list:
+        """
+
+        Args:
+            table_name: name of the table to select from
+            connector: Connector used to connect conditions. Must be suported by the DB
+            condition: conditions, connected by "equals"
+            skipdouble_dash_test: skip the crude SQLInjection test if it breaks your request,
+                    implement your OWN CHECK!
+
+        Returns: a list containing the rows returned
+
+        """
+        if await self._has_table(table_name):
+            if not condition:
+                condition = {}
+           # sql_string = "SELECT * FROM {table_name} WHERE ?;".format(table_name=table_name)
+            cond_str = ""
+            for k, v in condition.items():
+                cond_str += f"{k} = '{v}' {connector}"
+            cond_str = cond_str[0:-len(connector) - 1]
+            if ("--" in cond_str) and not skipdouble_dash_test:
+                raise ValueError("Suspicion of SQL-Injection. Aborting")
+            return self.cursor.execute(f"SELECT * FROM {table_name} WHERE {cond_str};").fetchall()
+        else:
+            raise ValueError(f"Table {table_name} does not exists!")
+
+    async def _has_table(self, name: str) -> bool:
+        """
+        checks whether the table exists.

```

Comment:
please update all docstrings to the Google Format.

on a side note, you *may* want to change pycharm / your IDE to generate Google Docstrings rather than the default RST ones